.phan/config.php

```diff
@@ -1,20 +1,33 @@
 <?php
 
 return [
-	"backward_compatibility_checks" => true,
```

Comment:
We're firmly on PHP 7 now so I don't think we need to check this.

Reply:
I believe this enables backwards compatibility checks against target_php_version, not against PHP7.

Reply:
The references I can find seem to specify PHP 5 --> PHP 7 changes. I don't think it has to do with target_php_version

https://github.com/phan/phan/wiki/Tutorial-for-Analyzing-a-Large-Sloppy-Code-Base#slowly-ramp-up-the-strength-of-the-analysis

https://github.com/phan/phan/wiki/Phan-Config-Settings#backward_compatibility_checks

Reply:
from the 2nd link:
```
Backwards Compatibility Checking. This is slow and expensive, but you should consider running it before upgrading your version of PHP to a new version that has backward compatibility breaks
```

Reading this, it is now specified for php 7 or 5.  We might have php8 in the future when thing break with php7.  Even if this possibility is low.  
But as @johnsaigle mentionned, all the code is done for php7 and the cost of this is high. WIll help with time to check.

Reply:
Even if it's "high cost", high is at most on the order of seconds for phan, in a CI run that takes over 20 minutes to run the integration tests. I think we should still keep it.